src/Uno.UWP/Helpers/StartStopEventWrapper.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+
+namespace Uno.Helpers
+{
+	internal class StartStopEventWrapper<TDelegate>
+		where TDelegate : Delegate
+	{
+		private readonly object _syncLock;
+		private readonly Action _onFirst;
+		private readonly Action _onLast;
+
+		/// <summary>
+		/// Creates a wrapper around an event, which needs to be synchronized
+		/// and needs to run an action when first subscriber is added and when
+		/// last subscriber is removed.
+		/// </summary>
+		/// <param name="onFirst">Action to run when first subscriber is added.</param>
```

Comment:
Maybe add a warning that this action will be run inside the lock and shouldn't invoke blocking operations etc.